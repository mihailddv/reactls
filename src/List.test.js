// Компонент List - список
// Напишите для него тесты.
// Компонент позволяет добавлять элементы в список.
// При добавлении нового элемента тексовое поле очищается.

// При клике на элементы списка они удаляются.

// Убедитесь, что вы протестировали всю функциональнось.

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { List } from './List';

describe('List', () => {

  describe('Component render correctly', () => {

    it('renders correctly', () => {
      const tree = renderer.create(<List />).toJSON();

      expect(tree).toMatchSnapshot();
    });

  });


  describe('Item work correctlly', () => {
    const container = shallow(<List />);
    const item = 'Text';

    it('add item', () => {      
      const input = container.find('.t-input');

      input.simulate('change', {
        target: { 
          value: item
        } 
      });

      container.find('.t-btn-add').simulate('click');

      expect(container.find('.t-list').contains(item)).toEqual(true);
    });


    it('input after add item is clear', () => {
      const input = container.find('.t-input');
      const inputValue = input.prop('value');

      expect(inputValue).toEqual('');
    });
    

    it('remove item', () => {
      container.find('.t-list__item').first().simulate('click');
      expect(container.find('.t-list__item').length).toEqual(0);
    });

  });

});