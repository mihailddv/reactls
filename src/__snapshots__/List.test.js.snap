// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component render correctly renders correctly 1`] = `
Array [
  <ul
    className="t-list"
  />,
  <input
    className="t-input"
    onChange={[Function]}
    value=""
  />,
  <button
    className="t-btn-add"
    onClick={[Function]}
  >
    Add item
  </button>,
]
`;
